idl/github.com/uber/cadence/shared.thrift

```diff
@@ -1333,6 +1340,12 @@ struct RespondQueryTaskCompletedRequest {
   40: optional string errorMessage
 }
 
+struct WorkflowQueryResult {
+  10: optional QueryResultType resultType
+  20: optional binary queryAnswer
```

Comment:
`queryResult` to be consistent with the struct name?

Reply:
I thought about this but the result is really either an error or an answer. It is still a result if an error is returned from the query handler.

Reply:
I think its ok to say `result` even if its an error.. 

Another option is `Response`.. `WorkflowQueryResponse` and `queryResponse`.. I have not really seen answer being used in APIs in general.

Reply:
If its called result then there will be a type WorkflowQueryResult.result where result can be nil.

Also if we call it WorkflowQueryResponse this is weird because we suffix things with Response if they are responses to API calls. 

I think answer is the right term, because a query returning an error is also a successful completion of the query from the cadence point of view - so error can be seen as a result just as much as result can be.

Reply:
Ok, yeah. that's odd.. Makes sense then..